### PR TITLE
Get albums

### PR DESCRIPTION
**Pull request by tinoqq**
## Summary
Registered users can access the list of albums
## Trello Card
https://trello.com/c/H6uUjPsm

### CODE REVIEW

File: app/controllers/album.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+const jwt = require('../services/jwt'),
+  getUserForToken = require('./user').getUserForToken,
+  request = require('request-promise-native'),
+  errors = require('../errors');
+
+exports.albumsHost = 'https://jsonplaceholder.typicode.com';
```
Comment (molinajulian) [inline app/controllers/album.js]: Use environment var for this

File: app/controllers/album.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+exports.albumsPath = '/albums';
```
Comment (molinajulian) [inline app/controllers/album.js]: same

File: app/controllers/album.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+const getAlbums = () => {
+  return request({ uri: exports.albumsHost + exports.albumsPath, json: true }).catch(e => {
```
Comment (molinajulian) [inline app/controllers/album.js]: Remove brackets and return

Comment (molinajulian) [inline app/controllers/album.js]: Move this for service. Use request promise

File: test/album.js [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+  itemsPerPage = Number.parseInt(process.env.DEFAULT_ITEMS_PER_PAGE),
+  testHelpers = require('./testHelpers'),
+  nock = require('nock'),
+  albumsController = require('../app/controllers/album');
+
+chai.use(chaiSubset);
+
+nock(albumsController.albumsHost)
+  .get(albumsController.albumsPath)
+  .replyWithFile(200, `${__dirname}/albums.json`, { 'Content-Type': 'application/json' })
```
Comment (molinajulian) [inline test/album.js]: Use only reply. Pass code and message. Is unnecessary create a file for this

File: test/album.js [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+  validationErrorMsgs = require('./../app/middlewares/user').validationErrorMessages,
+  jwt = require('../app/services/jwt'),
+  itemsPerPage = Number.parseInt(process.env.DEFAULT_ITEMS_PER_PAGE),
+  testHelpers = require('./testHelpers'),
+  nock = require('nock'),
+  albumsController = require('../app/controllers/album');
+
+chai.use(chaiSubset);
+
+nock(albumsController.albumsHost)
```
Comment (molinajulian) [inline test/album.js]: Nock uri

File: test/testHelpers.js [hunk 6]
```diff
@@ -0,0 +1,62 @@
+const chai = require('chai'),
```
Comment (molinajulian) [inline test/testHelpers.js:1]: Good Job for this :)

File: test/album.js [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+              expect(album).to.have.property('userId');
+              expect(album).to.have.property('id');
+              expect(album).to.have.property('title');
+            });
+
+            res.should.have.status(200);
+            done();
+          })
+      )
+    );
```
Comment (molinajulian) [inline test/album.js:115]: Add test for when the request file. Remember before or after test, nock must be clean. Check beforeEach or afterEach

**Review by molinajulian — CHANGES_REQUESTED** [review]

File: test/album.js (hunk 5 above)
Comment (tinoqq) [reply test/album.js]: ???

File: app/services/albums.js [hunk 8]
```diff
@@ -0,0 +1,3 @@
+const request = require('request-promise-native');
+
+exports.getAlbums = () => request({ uri: process.env.ALBUMS_HOST + process.env.ALBUMS_PATH, json: true });
```
Comment (molinajulian) [inline app/services/albums.js]: Catch error here, and in the controller in the catch call next()

**Review by molinajulian — CHANGES_REQUESTED** [review]

File: app/controllers/user.js [hunk 9]
```diff
@@ -27,7 +27,6 @@ exports.logIn = ({ user }, res, next) => {
       }
     })
     .catch(next);
-};
```
Comment (molinajulian) [inline app/controllers/user.js]: Why exist two logins? Delete one

**Review by molinajulian — CHANGES_REQUESTED** [review]

**Review by molinajulian — APPROVED** [review]

**Pull request merged**